Fix sorting algorithm

tyger opened the pull request:
Current approach for sorting process folder by folder. So that files are sorted within each folder independently so that if dependent files are in subfolder it going to fail to sort them in the right order.
This fix changes logic to first read all required files recursively and then sort them with `avrohugger-filesorter`.

makubi reviewed (changes requested):
Hi @tyger, thanks for your PR! It looks good to me. Could you squash the commits before merging?

The test works for me even without the fix, but I guess this is due to different ordering of files. I will open an issue for this to be fixed.

Thanks.

makubi commented:
Please have a look at https://github.com/makubi/avrohugger-maven-plugin/commit/74621a9433a31f4645fbac9dabf1402eace6afb5. What do you think?

tyger commented:
Hi @makubi, thanks for reviewing it.

tyger commented:
np, I will squash commits.
Let me double check the test, it was fine when I was experimenting, but maybe I changed something.

What comes to your proposed commit I think it makes sense only if you expecting alternative implementations of file lister to be provided by developers. Maybe it is not needed at this stage - I don't see right now what that alternative may be. However this flexibility might be useful in the future. 
Do you want me to add this to my PR?

tyger commented:
@makubi I've updated PR with suggested code.
Also I've checked the test that I had added originally and it did fail on the old version of sorter, so at least in Windows the behaviour is reproducible.

makubi reviewed (changes requested):
Hi @tyger, thanks for updating the PR and checking the tests! If you prefer, you can just remove the `FileListHelper` as you mentioned and I will have a look at it afterwards.

Sorry for the late reply. Could you please once again squash the commits? I will have a look at it afterwards and approve it if not other changes are introduced.

Thanks

tyger commented:
@makubi decided to keep `FileListHelper` to not introduce another round of changes. 
Commits are squashed. Please have a look again.

makubi reviewed (approved):
Thanks, looks good to me.
